harness/propertyHelper.js

```diff
@@ -7,7 +8,12 @@ function isConfigurable(obj, name) {
             $ERROR("Expected TypeError, got " + e);
         }
     }
-    return !Object.prototype.hasOwnProperty.call(obj, name);
+    var result = !Object.prototype.hasOwnProperty.call(obj, name);
+    if (result) {
+        // Put things back the way we found them.
+        Object.defineProperty(obj, name, desc);
```

Comment:
This call would fail when the property doesn't originally exist given that `desc` would be `undefined`. Instead of checking the result of the `hasOwnProperty` call in the `if` condition, we should check the descriptor.